src/commands.py

```diff
@@ -552,26 +564,24 @@ def __init__(self, *args, **kwargs):
 
     async def action(self, message):
         fields = []
-        show_mod = message.channel.id in (498634483910574082, 338237628275097601)
+        show_mod = message.channel.id in self.bot.config['mod_channels']
         try:
             cmd = message.content[1:].split()[1] # Tries to get the command to search for
         except IndexError:
-            for name, desc in self.commands_list.items():
-                if isinstance(desc, mod_only_command) and not show_mod:
+            for helptxt in self.commands_list.values():
+                if helptxt.mod_only and not show_mod:
                     continue
                 fields.append({
-                    "name": name,
-                    "value": desc
+                    "name": helptxt.name,
+                    "value": helptxt.desc
                 })
 
         else: # If a command is given
             if cmd in self.commands_list:
-                desc = self.commands_list[cmd]
-                if isinstance(desc, mod_only_command) and not show_mod:
-                    desc = "Provided command is unavailable."
+                helptxt = self.commands_list[cmd]
                 fields.append({
```

Comment:
I think we should have help_text create the embed field instead. This way, we can completely remove commands that only mods should see if it's in a non-mod channel. i.e. normal users should have no indication that these commands exist.

Reply:
My bad, I misread this code

Reply:
should i stick with this implementation then? because i'm in the process of rewriting it so it returns a correct field and working around that.